src/goto-symex/complexity_limiter.cpp

```diff
@@ -0,0 +1,214 @@
+/*******************************************************************\
+
+Module: Symbolic Execution
+
+Author: John Dumbell
+
+\*******************************************************************/
+
+#include "complexity_limiter.h"
+#include "goto_symex_state.h"
+#include <cmath>
+
+complexity_limitert::complexity_limitert(
+  message_handlert &message_handler,
+  const optionst &options)
+  : log(message_handler)
+{
+  std::size_t limit = options.get_signed_int_option("symex-complexity-limit");
+  if((complexity_active = limit > 0))
+  {
+    // This gives a curve that allows low limits to be rightly restrictive,
+    // while larger numbers are very large.
+    max_complexity = static_cast<std::size_t>((limit * limit) * 25);
+
+    const std::size_t failed_children_limit =
```

Comment:
child_loops? Also I guess this is an arbitrary complexity number rather than a count? If so use `_complexity` to indicate this isn't a count